bdns/dns_test.go

```diff
@@ -190,13 +188,31 @@ func serveLoopResolver(stopChan chan bool) chan bool {
 			fmt.Println(err)
 		}
 	}()
-	return waitChan
+}
+
+func pollServer() {
+	var conn *dns.Conn
+	timeout := time.Duration(time.Millisecond * 200)
+	tries := 25
+
+	for i := 0; conn == nil; i++ {
```

Comment:
Actually, another option is just replacing `tries` and using a `context.Context.WithDeadline` and use that to say the server _must_ be up within some specific period. This would let you use a much simpler loop like so (where `ticker` is a `time.Ticker` that takes the place of the `time.Sleep`)

```
for {
  select {
    case <-ctx.Done():
      // failure code
    case <-ticker.C:
      // connection attempt code
      // if connection works just return from function
  }
}
```